jax/interpreters/pxla.py

```diff
@@ -61,144 +61,138 @@
 
 FLAGS = flags.FLAGS
 
-unsafe_map, map = map, safe_map
+unsafe_map, map = map, safe_map  # type: ignore
 
 Index = Union[int, slice, Tuple[Union[int, slice], ...]]
 
 
-def device_put(x, devices: Sequence[xb.xla_client.Device], replicate: bool=False) -> List[xb.xla_client._xla.PyLocalBuffer]:
-  """Call device_put on a sequence of devices and return a flat sequence of buffers."""
-  if replicate:
-    return list(it.chain.from_iterable(xla.device_put(x, device) for device in devices))
-  else:
-    return list(it.chain.from_iterable(xla.device_put(val, device) for val, device in safe_zip(x, devices)))
+class Unstacked(NamedTuple):
+  size: int
+
+class Sharded(NamedTuple):
```

Comment:
My preference would be `Tiled`, but I figured that it's part of a `ShardingSpec`, so `Sharded` might be the preferred term here. `Partitioned` makes it sound as if it was an XLA-partitioner specific thing (which it is _for now_, because that's the only place where we use it at the moment, but it might not always be like that). Also, `Partitioned` implies a little bit that it's been spread across a number of devices, but the purpose of `.sharding` is only to describe how to split up the value into chunks, abstracting over the device assignment. Hence, I'd lean towards `Sharded` or `Tiled`.

But I also don't care about the particular names too much, so I can change it if you feel strongly about it.

Reply:
Ok, let's do `Partitioned`.

Reply:
I think I like `Chunked`. Let's do that.